velox/type/DecimalUtil.h

```diff
@@ -32,5 +32,44 @@ class DecimalUtil {
   /// Helper function to convert a decimal value to string.
   template <typename T>
   static std::string toString(const T& value, const TypePtr& type);
+
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleWithRoundUp(
+      const TInput inputValue,
+      const int fromPrecision,
+      const int fromScale,
+      const int toPrecision,
+      const int toScale,
+      const bool nullOnFailure) {
+    TOutput rescaledValue(inputValue.unscaledValue());
```

Comment:
`TInput = LongDecimal, TOutput = ShortDecimal` will cause a problem here. We have to use int128_t I suppose.